frontend/src/v5/store/federations/federations.sagas.ts

```diff
@@ -97,6 +105,64 @@ export function* fetchFederationStats({ teamspace, projectId, federationId }: Fe
 	}
 }
 
+export function* fetchFederationViews({
+	teamspace,
+	projectId,
+	federationId,
+}: FetchFederationViewsAction) {
+	try {
+		const { views }: FetchFederationViewsResponse = yield API.Federations.fetchFederationViews({
+			teamspace,
+			projectId,
+			federationId,
+		});
+		yield put(FederationsActions.fetchFederationViewsSuccess(projectId, federationId, views));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: 'trying to fetch federation views',
+			error,
+		}));
+	}
+}
+
+export function* fetchFederationSettings({
+	teamspace,
+	projectId,
+	federationId,
+}: FetchFederationSettingsAction) {
+	try {
+		const rawSettings = yield API.Federations.fetchFederationSettings({
+			teamspace,
+			projectId,
+			federationId,
+		});
+		const settings = prepareFederationSettingsForFrontend(rawSettings);
+		yield put(FederationsActions.fetchFederationSettingsSuccess(projectId, federationId, settings));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: 'trying to fetch federation settings',
+			error,
+		}));
+	}
+}
+
+export function* updateFederationSettings({
+	teamspace, projectId, federationId, settings, extraSettings,
+}: UpdateFederationSettingsAction) {
+	try {
+		const rawSettings = prepareFederationSettingsForBackend(settings, extraSettings);
+		yield API.Federations.updateFederationSettings({
+			teamspace, projectId, federationId, rawSettings,
+		});
+		yield put(FederationsActions.updateFederationSettingsSuccess(projectId, federationId, settings, extraSettings));
```

Comment:
Same question for updateFederationSettingsSuccess .why we have separate fields for settings?